File: Prueba1/app/src/main/java/com/cafeConPalito/prueba1/MainActivity2.java
package com.cafeConPalito.prueba1;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

public class MainActivity2 extends AppCompatActivity {

    private TextView tv4;
    private Button button;

    private ImageView iv2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        iv2 = findViewById(R.id.iv2);
        iv2.setImageResource(R.drawable.logo);

        tv4= findViewById(R.id.tv4);

        Bundle recibedatos = getIntent().getExtras();
        String nombre = recibedatos.getString("nombre");

        tv4.setText("Bienvenido, " + nombre);

    }
    private void salir(View v){
        finish();
    }

}

File: PruebaRamiro/app/src/main/java/com/cafeconpalito/pruebaramiro/MainActivity2.kt
package com.cafeconpalito.pruebaramiro

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.ImageView

class MainActivity2 : AppCompatActivity() {

    lateinit var ivDoubleCheck: ImageView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        initUi()
    }
    private fun initVariables() {
        ivDoubleCheck = findViewById(R.id.ivDoubleCheck)
    }
    private fun initUi() {
        initVariables()
        initListeners()
    }
    private fun initListeners() {
        ivDoubleCheck.setOnClickListener {changeView()}
    }

    private fun changeView() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }
}